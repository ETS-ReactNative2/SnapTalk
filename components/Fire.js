import FirebaseKeys from "./config";
import firebase from 'firebase';
require("firebase/firestore");

class Fire {

    constructor() {
        firebase.initializeApp(FirebaseKeys);
    }


    addPost = async ({ text, localUri, postKey }) => {
        const remoteUri = await this.uploadPhotoAsync(localUri, 'photos/' + this.uid + '/' + Date.now());

        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathName = new firebase.firestore.FieldPath('name');

        const userAgain = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathProfilePicture = new firebase.firestore.FieldPath('profilePicture');

        return new Promise((res, rej) => {
            this.firestore.collection("posts").add({
                text,
                uid: this.uid,
                postKey: '',
                timestamp: this.timestamp,
                image: remoteUri,
                username: user.get(fieldPathName),
                avatar: userAgain.get(fieldPathProfilePicture),
                listOfComments: [],
                nbOfComments: 0,
                listOfLikes: [],
                nbOfLikes: 0,
            })
                .then(ref => {
                    res(ref);
                })
                .catch(error => {
                    rej(error);
                })
        })
    };

    updateUserLikedList = async (postId) => {
        let db = this.firestore.collection("posts").doc(postId);

        const post = await firebase.firestore().collection("posts").doc(postId).get();
        const fieldPathListOfLikes = new firebase.firestore.FieldPath('listOfLikes');
        const arrayOfLikes = await post.get(fieldPathListOfLikes)

        if (arrayOfLikes.includes(this.uid)) {
            db.update({
                listOfLikes: firebase.firestore.FieldValue.arrayRemove(this.uid),
            })
        }
        else {
            db.update({
                listOfLikes: firebase.firestore.FieldValue.arrayUnion(this.uid),
            })
        }
    }



    addPostKey = async (postId) => {

        let dbPost = this.firestore.collection("posts").doc(postId);

        if (postId) {
            dbPost.update({
                postKey: postId
            })
        }
    };

    updatePostList = async (postId) => {
        let db = this.firestore.collection("users").doc(this.uid);

        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathListOfPosts = new firebase.firestore.FieldPath('listOfPosts');

        if (postId) {
            db.update({
                listOfPosts: firebase.firestore.FieldValue.arrayUnion(postId)
            })

            const nbOfPosts = await user.get(fieldPathListOfPosts).length + 1;

            db.set(
                { nbOfPosts: nbOfPosts }, { merge: true }
            )
        }
    }

    addComment = async ({ text, postKey }) => {
        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathName = new firebase.firestore.FieldPath('name');

        const userAgain = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathProfilePicture = new firebase.firestore.FieldPath('profilePicture');

        return new Promise((res, rej) => {
            this.firestore.collection("comments").add({
                comment: text,
                uid: this.uid,
                timestamp: this.timestamp,
                username: user.get(fieldPathName),
                postKey: postKey,
                commentKey: '',
                avatar: userAgain.get(fieldPathProfilePicture)
            })
                .then(ref => {
                    res(ref);
                })
                .catch(error => {
                    rej(error);
                })
        })
    };

    addCommentKey = async (commentId) => {

        let dbComment = this.firestore.collection("comments").doc(commentId);

        if (commentId) {
            dbComment.update({
                commentKey: commentId
            })
        }
    };

    updateCommentList = async ({ commentId, postId }) => {
        let dbUser = this.firestore.collection("users").doc(this.uid);

        let dbPost = this.firestore.collection("posts").doc(postId);

        const user = await firebase.firestore().collection("users").doc(this.uid).get();

        const fieldPathListOfComments = new firebase.firestore.FieldPath('listOfComments');

        if (commentId) {
            dbUser.update({
                listOfComments: firebase.firestore.FieldValue.arrayUnion(commentId)
            })

            const nbOfComments = await user.get(fieldPathListOfComments).length + 1;

            dbUser.set(
                { nbOfComments: nbOfComments }, { merge: true }
            )
        }

        if (postId) {
            dbPost.update({
                listOfComments: firebase.firestore.FieldValue.arrayUnion(commentId)
            })

            const nbOfComments = await user.get(fieldPathListOfComments).length + 1;

            dbPost.set(
                { nbOfComments: nbOfComments }, { merge: true }
            )
        }
    }

    createUser = async user => {
        let remoteAvatarUri = null

        try {
            if (!user.name) {
                throw new Error("Username was not entered.")
            }

            await this.firestore
                .collection("users")
                .get()
                .then(snapshot => {
                    snapshot.forEach(doc => {
                        if (user.name == doc.data().name) {
                            throw new Error("Username already taken.")
                        }
                    })
                })

            await firebase.auth().createUserWithEmailAndPassword(user.email, user.password);

            let db = this.firestore.collection("users").doc(this.uid);

            db.set({
                name: user.name,
                email: user.email,
                listOfFollowers: [],
                listOfFollowing: [],
                listOfPosts: [],
                listOfComments: [],
                nbOfPosts: 0,
                nbOfComments: 0,
                profilePicture: remoteAvatarUri
            });

            if (user.avatar) {
                remoteAvatarUri = await this.uploadPhotoAsync(user.avatar, 'avatar/' + this.uid);
                db.set({ profilePicture: remoteAvatarUri }, { merge: true })
            }

        } catch (error) {
            alert("Error: " + error.message);
        }
    }

    uploadPhotoAsync = async (uri, filename) => {

        return new Promise(async (res, rej) => {
            const response = await fetch(uri);
            const file = await response.blob();

            let upload = firebase
                .storage()
                .ref(filename)
                .put(file);

            upload.on("state_changed", snapshot => { }, err => {
                rej(err);
            },
                async () => {
                    const url = await upload.snapshot.ref.getDownloadURL();
                    res(url);
                }
            );
        });
    };


    get firestore() {
        return firebase.firestore();
    }

    get uid() {
        return (firebase.auth().currentUser || {}).uid;
    }

    get timestamp() {
        return Date.now();
    }
}


Fire.shared = new Fire();
export default Fire;