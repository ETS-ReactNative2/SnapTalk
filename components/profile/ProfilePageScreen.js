import React, { Component } from 'react';
import { View, Text, Image, ImageBackground, FlatList, Modal, Button } from 'react-native';
import Fire from '../firebase/Fire';
import LogoutButton from '../profile/LogoutButton';
import firebase from 'firebase';
import Post from '../post/Post';
import PicColor from './PicColor';
import styles from '../profile/style/ProfilePageScreenStyle';

export default class ProfilePageScreen extends Component {

  state = {
    user: {},
    nbOfFollowers: 0,
    nbOfFollowing: 0,
    nbOfPosts: 0,
    posts: [],
    isLoading: false,
    postInArray: false,
    infoColor: "#EFECF4",
    isModalVisible: false
  };

  unsubscribe = null;

  componentDidMount() {
    this.getData();
  }

  getData = async () => {
    const user = this.props.uid || Fire.shared.uid;

    this.unsubscribe = Fire.shared.firestore
      .collection('users')
      .doc(user)
      .onSnapshot(doc => {
        this.setState({ user: doc.data() });
      });

    this.getListSize();
    this.getCurrentUserPost();
  };

  getListSize = async () => {
    const user = await firebase
      .firestore()
      .collection('users')
      .doc(Fire.shared.uid)
      .get();

    const listOfPosts = new firebase.firestore.FieldPath('listOfPosts');

    this.setState({ nbOfPosts: await user.get(listOfPosts).length });

    const listOfFollowers = new firebase.firestore.FieldPath('listOfFollowers');

    this.setState({ nbOfFollowers: await user.get(listOfFollowers).length });

    const listOfFollowing = new firebase.firestore.FieldPath('listOfFollowing');

    this.setState({ nbOfFollowing: await user.get(listOfFollowing).length });
  };

  changeModalVisibility = (bool) => {
    this.setState({ isModalVisible: bool });
  }

  setColor = (data) => {
    this.setState({ infoColor: data });
  }

  getCurrentUserPost = async () => {
    this.setState({ isLoading: true });

    this.setState({ posts: [] })

    Fire.shared.firestore
      .collection('posts')
      .get()
      .then(snapshot => {
        snapshot.forEach(doc => {
          this.setState({ postInArray: false });

          if (Fire.shared.uid == doc.data().uid) {
            this.state.posts.push(doc.data());
          }
        });

        this.setState({ posts: this.state.posts.sort(function (a, b) { return (parseInt(b.timestamp) - parseInt(a.timestamp)) }) })
      })
      .finally(() => this.setState({ isLoading: false }));
  };

  renderPost = post => {
    return <Post post={post} profilePagePost={true} />;
  };

  profileHeaderRender = () => {
    return (
      <View style={{ backgroundColor: this.state.infoColor }} >
        <View style={styles.header}>
          <Button title="Change Color" color="purple" onPress={() => this.changeModalVisibility(true)}>
            <Text>Open Modal</Text>
          </Button>

          <Modal visible={this.state.isModalVisible} onRequestClose={() => this.changeModalVisibility(false)}>
            <PicColor changeModalVisibility={this.changeModalVisibility} setColor={this.setColor} />
          </Modal>
          <LogoutButton />
        </View>

        <View>
          <View styles={styles.container}>
            <View style={styles.imageContainer}>
              <ImageBackground source={require('../../assets/Default-profile-bg.jpg')} style={styles.imageBackground}>
                <View style={styles.avatarContainer}>
                  <Image style={styles.avatar} source={this.state.user.profilePicture ? { uri: this.state.user.profilePicture } : require('../../assets/tempAvatar.jpg')}></Image>
                </View>

                <Text style={styles.name}> {this.state.user.name} </Text>
              </ImageBackground>
            </View>

            <View style={styles.info}>
              <View style={styles.state}>
                <Text style={styles.amount}> {this.state.nbOfPosts} </Text>
                <Text style={styles.title}> Posts </Text>
              </View>

              <View style={[styles.state, { borderColor: "#DFD8C8", borderLeftWidth: 1, borderRightWidth: 1 }]}>
                <Text style={styles.amount}> {this.state.nbOfFollowers} </Text>
                <Text style={styles.title}> Followers </Text>
              </View>

              <View style={styles.state}>
                <Text style={styles.amount}> {this.state.nbOfFollowing} </Text>
                <Text style={styles.title}> Following </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    );
  };

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          style={styles.feed}
          data={this.state.posts}
          renderItem={({ item }) => this.renderPost(item)}
          ListHeaderComponent={this.profileHeaderRender}
          keyExtractor={item => item.id}
          showsVerticalScrollIndicator={false}
          refreshing={this.state.isLoading}
          onRefresh={this.getData}
        />
      </View>
    );
  }
}