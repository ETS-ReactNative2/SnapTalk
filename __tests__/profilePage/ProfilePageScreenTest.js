import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProfilePageScreen from '../../components/profile/ProfilePageScreen'
import Fire from '../../components/firebase/Fire'

describe('ProfilePageScreen Test', () => {
    let profilePageScreenSnapshot = renderer.create(<ProfilePageScreen />).toJSON();

    test('ProfilePageScreen Render Test', () => {
        expect(profilePageScreenSnapshot).toMatchSnapshot();
    })
})