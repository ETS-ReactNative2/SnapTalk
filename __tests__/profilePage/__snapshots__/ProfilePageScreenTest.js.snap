// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePageScreen Test ProfilePageScreen Render Test 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#EFECF4",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-end",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "purple",
                      },
                    ]
                  }
                >
                  Change Color
                </Text>
              </View>
            </View>
            <Modal
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "backgroundColor": "white",
                        "borderRadius": 10,
                        "height": 100,
                        "paddingTop": 10,
                        "textAlign": "center",
                      },
                      Object {
                        "height": 1254,
                        "width": 670,
                      },
                    ]
                  }
                >
                  <RCTScrollView
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  >
                    <View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Red
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Blue
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Yellow
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Orange
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Green
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Gray
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Black
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Purple
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Pink
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Brown
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Navy
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Fuchsia
                        </Text>
                        <View />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "stretch",
                            "backgroundColor": "white",
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                          }
                        >
                          Silver
                        </Text>
                      </View>
                    </View>
                  </RCTScrollView>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "alignSelf": "stretch",
                          "backgroundColor": "white",
                          "borderRadius": 10,
                          "flex": 1,
                          "paddingVertical": 10,
                        },
                        Object {
                          "backgroundColor": "orange",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "margin": 5,
                        }
                      }
                    >
                      Choose color
                    </Text>
                  </View>
                </View>
              </View>
            </Modal>
            <View
              accessibilityRole="button"
              accessibilityStates={Array []}
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "padding": 8,
                        "textAlign": "center",
                      },
                      Object {
                        "color": "purple",
                      },
                    ]
                  }
                >
                  Logout
                </Text>
              </View>
            </View>
          </View>
          <View>
            <View
              styles={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "paddingBottom": 10,
                  }
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#52575D",
                      "borderTopWidth": 1,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../assets/Default-profile-bg.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        Object {
                          "height": undefined,
                          "width": undefined,
                        },
                        undefined,
                      ]
                    }
                  />
                  <View
                    style={
                      Object {
                        "paddingTop": 10,
                        "shadowColor": "#151734",
                        "shadowOpacity": 0.4,
                        "shadowRadius": 30,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../assets/tempAvatar.jpg",
                        }
                      }
                      style={
                        Object {
                          "borderColor": "#6495ED",
                          "borderRadius": 75,
                          "borderWidth": 5,
                          "height": 150,
                          "width": 150,
                        }
                      }
                    />
                  </View>
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "marginTop": 24,
                        "paddingBottom": 10,
                      }
                    }
                  >
                     
                     
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#FFF",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "margin": 32,
                    "marginVertical": 8,
                    "padding": 8,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#52575D",
                        "fontFamily": "HelveticaNeue",
                        "fontSize": 18,
                      }
                    }
                  >
                     
                    0
                     
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#AEB5BC",
                        "fontSize": 12,
                        "fontWeight": "500",
                        "marginTop": 4,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                     Posts 
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      Object {
                        "borderColor": "#DFD8C8",
                        "borderLeftWidth": 1,
                        "borderRightWidth": 1,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#52575D",
                        "fontFamily": "HelveticaNeue",
                        "fontSize": 18,
                      }
                    }
                  >
                     
                    0
                     
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#AEB5BC",
                        "fontSize": 12,
                        "fontWeight": "500",
                        "marginTop": 4,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                     Followers 
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#52575D",
                        "fontFamily": "HelveticaNeue",
                        "fontSize": 18,
                      }
                    }
                  >
                     
                    0
                     
                  </Text>
                  <Text
                    style={
                      Object {
                        "color": "#AEB5BC",
                        "fontSize": 12,
                        "fontWeight": "500",
                        "marginTop": 4,
                        "textTransform": "uppercase",
                      }
                    }
                  >
                     Following 
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
